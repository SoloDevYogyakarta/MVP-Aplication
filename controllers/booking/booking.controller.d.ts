import { Request, Response } from 'express';
import { DyanmicQuery } from '../../validators/query/product.query';
import { BookingRepository } from '../../repository/booking/booking.repository';
import { BookingService } from '../../services/booking/booking.service';
import { BookingField } from '../../validators/booking/booking.validator';
export declare class BookingController {
    private readonly repository;
    private readonly service;
    constructor(repository: BookingRepository, service: BookingService);
    created(res: Response, body: BookingField): Promise<Response<any, Record<string, any>>>;
    list(query: DyanmicQuery, req: Request, res: Response): Promise<Response<any, Record<string, any>>>;
    detail(req: Request, res: Response): Promise<Response<any, Record<string, any>>>;
    destory(req: Request, res: Response): Promise<Response<any, Record<string, any>>>;
}
