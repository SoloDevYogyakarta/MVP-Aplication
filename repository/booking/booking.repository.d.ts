import { BookingInstance } from '../../database/entities/services/booking-entity/booking-entity';
import { DyanmicQuery } from '../../validators/query/product.query';
export declare class BookingRepository {
    findOne(public_id: string): Promise<BookingInstance>;
    findAll(query: DyanmicQuery): Promise<BookingInstance[]>;
}
