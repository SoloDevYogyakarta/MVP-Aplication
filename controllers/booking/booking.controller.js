"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingController = void 0;
const common_1 = require("@nestjs/common");
const guards_middleware_1 = require("../../middleware/guards.middleware");
const product_query_1 = require("../../validators/query/product.query");
const booking_repository_1 = require("../../repository/booking/booking.repository");
const booking_service_1 = require("../../services/booking/booking.service");
const booking_validator_1 = require("../../validators/booking/booking.validator");
let BookingController = class BookingController {
    constructor(repository, service) {
        this.repository = repository;
        this.service = service;
    }
    async created(res, body) {
        const result = await this.service.create(body);
        return res.status(common_1.HttpStatus.CREATED).json(result);
    }
    async list(query, req, res) {
        const result = await this.repository.findAll(query);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
    async detail(req, res) {
        const result = await this.repository.findOne(req.params.id);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
    async destory(req, res) {
        const result = await this.service.destroy(req.params.id);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
};
exports.BookingController = BookingController;
__decorate([
    (0, common_1.Post)(),
    __param(0, (0, common_1.Res)()),
    __param(1, (0, common_1.Body)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, booking_validator_1.BookingField]),
    __metadata("design:returntype", Promise)
], BookingController.prototype, "created", null);
__decorate([
    (0, common_1.Get)(),
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    __param(0, (0, common_1.Query)()),
    __param(1, (0, common_1.Req)()),
    __param(2, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [product_query_1.DyanmicQuery, Object, Object]),
    __metadata("design:returntype", Promise)
], BookingController.prototype, "list", null);
__decorate([
    (0, common_1.Get)(':id'),
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], BookingController.prototype, "detail", null);
__decorate([
    (0, common_1.Delete)(':id'),
    (0, common_1.UseGuards)(guards_middleware_1.AuthGuard),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], BookingController.prototype, "destory", null);
exports.BookingController = BookingController = __decorate([
    (0, common_1.Controller)('booking'),
    __metadata("design:paramtypes", [booking_repository_1.BookingRepository,
        booking_service_1.BookingService])
], BookingController);
//# sourceMappingURL=booking.controller.js.map