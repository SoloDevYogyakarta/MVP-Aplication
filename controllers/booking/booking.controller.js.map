{"version":3,"file":"booking.controller.js","sourceRoot":"","sources":["../../../src/controllers/booking/booking.controller.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;AACA,2CAQwB;AAGjB,IAAM,iBAAiB,GAAvB,MAAM,iBAAiB;IAE5B,OAAO,CAAQ,GAAY,EAAS,GAAa;QAC/C,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,OAAO,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,CAAC;IACxD,CAAC;IAGD,OAAO,CAAQ,GAAY,EAAS,GAAa;QAC/C,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,OAAO,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,CAAC;IACxD,CAAC;IAGD,IAAI,CAAQ,GAAY,EAAS,GAAa;QAC5C,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,EAAE,CAAC,CAAC,IAAI,CAAC,MAAM,CAAC,CAAC;IAChD,CAAC;IAGD,MAAM,CAAQ,GAAY,EAAS,GAAa;QAC9C,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,EAAE,CAAC,CAAC,IAAI,CAAC,QAAQ,CAAC,CAAC;IAClD,CAAC;IAGD,OAAO,CAAQ,GAAY,EAAS,GAAa;QAC/C,OAAO,GAAG,CAAC,MAAM,CAAC,mBAAU,CAAC,EAAE,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,CAAC;IACnD,CAAC;CACF,CAAA;AAzBY,8CAAiB;AAE5B;IADC,IAAA,aAAI,GAAE;IACE,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;gDAElC;AAGD;IADC,IAAA,aAAI,EAAC,KAAK,CAAC;IACH,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;gDAElC;AAGD;IADC,IAAA,YAAG,GAAE;IACA,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;6CAE/B;AAGD;IADC,IAAA,YAAG,EAAC,KAAK,CAAC;IACH,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;+CAEjC;AAGD;IADC,IAAA,eAAM,EAAC,KAAK,CAAC;IACL,WAAA,IAAA,YAAG,GAAE,CAAA;IAAgB,WAAA,IAAA,YAAG,GAAE,CAAA;;;;gDAElC;4BAxBU,iBAAiB;IAD7B,IAAA,mBAAU,EAAC,SAAS,CAAC;GACT,iBAAiB,CAyB7B"}