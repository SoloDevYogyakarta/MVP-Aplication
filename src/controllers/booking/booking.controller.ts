import { Request, Response } from 'express';
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpStatus,
  Post,
  Query,
  Req,
  Res,
  UseGuards,
} from '@nestjs/common';
import { AuthGuard } from '../../middleware/guards.middleware';
import { DyanmicQuery } from '../../validators/query/product.query';
import { BookingRepository } from '../../repository/booking/booking.repository';
import { BookingService } from '../../services/booking/booking.service';
import { BookingField } from '../../validators/booking/booking.validator';

@Controller('booking')
export class BookingController {
  constructor(
    private readonly repository: BookingRepository,
    private readonly service: BookingService,
  ) {}

  @Post()
  async created(@Res() res: Response, @Body() body: BookingField) {
    const result = await this.service.create(body);
    return res.status(HttpStatus.CREATED).json(result);
  }

  @Get()
  @UseGuards(AuthGuard)
  async list(
    @Query() query: DyanmicQuery,
    @Req() req: Request,
    @Res() res: Response,
  ) {
    const result = await this.repository.findAll(query);
    return res.status(HttpStatus.OK).json(result);
  }

  @Get(':id')
  @UseGuards(AuthGuard)
  async detail(@Req() req: Request, @Res() res: Response) {
    const result = await this.repository.findOne(req.params.id);
    return res.status(HttpStatus.OK).json(result);
  }

  @Delete(':id')
  @UseGuards(AuthGuard)
  async destory(@Req() req: Request, @Res() res: Response) {
    const result = await this.service.destroy(req.params.id);
    return res.status(HttpStatus.OK).json(result);
  }
}
