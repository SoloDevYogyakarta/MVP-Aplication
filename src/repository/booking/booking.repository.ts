import { Injectable } from '@nestjs/common';
import {
  bookingEntity,
  BookingInstance,
} from '../../database/entities/services/booking-entity/booking-entity';
import { dynamicFilter } from '../../utils/dynamic-filter/dynamic-filter';
import { DyanmicQuery } from '../../validators/query/product.query';

@Injectable()
export class BookingRepository {
  async findOne(public_id: string): Promise<BookingInstance> {
    return await bookingEntity.findOne({ where: { public_id } });
  }

  async findAll(query: DyanmicQuery): Promise<BookingInstance[]> {
    let where = {},
      datas = [];
    where = dynamicFilter(where, datas, query);
    return await bookingEntity.findAll({
      where: where,
    });
  }
}
