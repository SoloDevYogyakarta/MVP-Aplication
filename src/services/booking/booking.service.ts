import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import { BookingField } from '../../validators/booking/booking.validator';
import { bookingEntity } from '../../database/entities/services/booking-entity/booking-entity';
import { omit } from 'lodash';
import { nanoid } from 'nanoid';
import { createpath } from '../../utils/system/system';

@Injectable()
export class BookingService {
  private readonly regex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g;

  async create(body: BookingField) {
    const data = { username: body.token };
    if (this.regex.test(body.token)) {
      data['email'] = body.token;
      delete data.username;
    }
    const create = await bookingEntity.create({
      public_id: nanoid(),
      ...data,
      ...omit(body, ['token']),
    });
    create.save();
    createpath('../../database/dataTxt/booking-http-entity.txt', create);
    return { status: HttpStatus.CREATED, message: 'Booking has been create' };
  }

  async destroy(public_id: string) {
    const findOne = await bookingEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.NOT_FOUND,
          message: 'Booking not found',
        },
        HttpStatus.NOT_FOUND,
      );
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Booking has been delete' };
  }
}
