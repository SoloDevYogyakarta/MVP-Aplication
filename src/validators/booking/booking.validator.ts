import { IsDate, IsOptional, IsString } from 'class-validator';

export class BookingField {
  @IsString()
  token: string;

  @IsDate()
  day!: Date;

  @IsString()
  @IsOptional()
  desc!: string;
}
