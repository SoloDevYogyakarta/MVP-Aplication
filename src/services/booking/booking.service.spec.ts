import { BookingService } from './booking.service';

describe('BookingService', () => {
  let service: BookingService;

  beforeEach(() => {
    service = new BookingService();
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('should create the app', () => expect(service).toMatchSnapshot());
});
