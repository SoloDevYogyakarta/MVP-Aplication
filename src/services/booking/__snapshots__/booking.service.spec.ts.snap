// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingService should create the app 1`] = `
BookingService {
  "regex": /\\^\\[\\\\w-\\\\\\.\\]\\+@\\(\\[\\\\w-\\]\\+\\\\\\.\\)\\+\\[\\\\w-\\]\\{2,4\\}\\$/g,
}
`;
